presto-native-execution/presto_cpp/main/LinuxMemoryChecker.cpp

```diff
@@ -49,6 +49,78 @@ class LinuxMemoryChecker : public PeriodicMemoryChecker {
     LOG(INFO) << fmt::format("Changed to using memory stat file {}", statFile_);
   }
 
+  void setMemMaxFile(std::string memMaxFile) {
+    memMaxFile_ = memMaxFile;
+    LOG(INFO) << fmt::format(
+        "Changed to using memory max file {}", memMaxFile_);
+  }
+
+  void setMemInfo(std::string memInfoFile) {
+    memInfoFile_ = memInfoFile;
+    LOG(INFO) << fmt::format("Changed to using meminfo file {}", memInfoFile_);
+  }
+
+  void start() {
+    // Set memMaxFile_ to:
+    // "/sys/fs/cgroup/memory/memory.limit_in_bytes" for cgroup v1
+    // or
+    // "/sys/fs/cgroup/memory.max" for cgroup v2.
+    struct stat buffer;
+    if ((stat(kCgroupV1MaxMemFilePath, &buffer) == 0)) {
+      memMaxFile_ = kCgroupV1MaxMemFilePath;
+    } else if ((stat(kCgroupV2MaxMemFilePath, &buffer) == 0)) {
+      memMaxFile_ = kCgroupV2MaxMemFilePath;
+    } else {
+      memMaxFile_ = "None";
+    }
+    LOG(INFO) << fmt::format("Checking {}", memMaxFile_);
+
+    // Check system-memory-gb < system-mem-limit-gb < actual total memory
+    // capacity.
+    VELOX_CHECK_LE(
+        config_.systemMemLimitBytes,
+        getActualTotalMemory(),
+        "system-mem-limit-gb is higher than the actual total memory capacity.");
+
+    auto* systemConfig = SystemConfig::instance();
+    if (config_.systemMemLimitBytes < systemConfig->systemMemoryGb()) {
+      LOG(WARNING) << "system-mem-limit-gb is smaller than system-memory-gb. "
+                   << "Expected: system-mem-limit-gb >= system-memory-gb.";
+    }
+
+    PeriodicMemoryChecker::start();
+  }
+
+  int64_t getActualTotalMemory() {
+    // Set actual total memory to be the smaller number between /proc/meminfo
```

Comment:
Should we add `// For cgroup v1:` in the comment as before?

Reply:
Both cgroup v1 and cgroup v2 will use the MemTotal number in /proc/meminfo if it is smaller than their respective memory.limit_in_bytes or memory.max value. 

Added more comments in the function for cgroup v1 to explain what's going on when extracting the value from memory.limit_in_bytes.